packages/contracts-periphery/contracts/foundry-tests/citizen-house/BadgeAdmin.t.sol

```diff
@@ -0,0 +1,692 @@
+            citizenAdrs[3] = mkadr("citizen4");
+        } else if (set == 1) {
+            citizenAdrs[0] = mkadr("citizen5");
+            citizenAdrs[1] = mkadr("citizen6");
+            citizenAdrs[2] = mkadr("citizen7");
+            citizenAdrs[3] = mkadr("citizen8");
+        } else if (set == 2) {
+            citizenAdrs[0] = mkadr("citizen9");
+            citizenAdrs[1] = mkadr("citizen10");
+            citizenAdrs[2] = mkadr("citizen11");
+            citizenAdrs[3] = mkadr("citizen12");
+        } else if (set == 3) {
+            citizenAdrs[0] = mkadr("citizen13");
+            citizenAdrs[1] = mkadr("citizen14");
+            citizenAdrs[2] = mkadr("citizen15");
+            citizenAdrs[3] = mkadr("citizen16");
+        } else if (set == 4) {
+            citizenAdrs[0] = mkadr("citizen17");
+            citizenAdrs[1] = mkadr("citizen18");
+            citizenAdrs[2] = mkadr("citizen19");
+            citizenAdrs[3] = mkadr("citizen20");
+        } else if (set == 5) {
+            citizenAdrs[0] = mkadr("citizen21");
+            citizenAdrs[1] = mkadr("citizen22");
+            citizenAdrs[2] = mkadr("citizen23");
+            citizenAdrs[3] = mkadr("citizen24");
+        }
+
+        return citizenAdrs;
+    }
+
+    function getSupplySet(uint256 set) public pure returns (uint256[] memory) {
+        uint256[] memory supply = new uint256[](4);
+        if (set == 0) {
+            supply[0] = 99;
+            supply[1] = 99;
+            supply[2] = 99;
+            supply[3] = 99;
+        } else if (set == 1) {
+            supply[0] = 10;
+            supply[1] = 11;
+            supply[2] = 12;
+            supply[3] = 14;
+        } else if (set == 2) {
+            supply[0] = 15;
+            supply[1] = 19;
+            supply[2] = 4;
+            supply[3] = 5;
+        } else if (set == 3) {
+            supply[0] = 5;
+            supply[1] = 5;
+            supply[2] = 3;
+            supply[3] = 2;
+        } else if (set == 4) {
+            supply[0] = 1;
+            supply[1] = 1;
+            supply[2] = 3;
+            supply[3] = 4;
+        } else if (set == 5) {
+            supply[0] = 15;
+            supply[1] = 15;
+            supply[2] = 15;
+            supply[3] = 15;
+        }
+        return supply;
+    }
+
+    function _basicSetup() public {
+        address[] memory opcoAdrs = getOPCOSet(0);
+        uint256[] memory opcoSupply = getSupplySet(0);
+        address[] memory citizenAdrs = getCitizenSet(0);
+
+        vm.prank(opAdr);
+        badgeAdmin.addOPCOs(opcoAdrs, opcoSupply);
+        vm.prank(opcoAdrs[0]);
+        badgeAdmin.addCitizens(citizenAdrs);
+
+        vm.prank(0xb4c79daB8f259C7Aee6E5b2Aa729821864227e84); // deployer address
+        badgeAdmin.updateBadgeContract(address(badge));
+        badge.updateAdminContract(address(badgeAdmin));
+    }
+
+    /** OP CONTROL */
+
+    function testOPControl() public {
+        address[] memory opcoAdrs = getOPCOSet(0);
```

Comment:
Perhaps split into 2 test contracts so that `_basicSetup` can just be `setUp` in one of the contracts and create a different test contract that doesn't have that if these tests that dont call basic setup require that its not called

Reply:
Good thinking! I pulled all relevant tests into separate contracts within `BadgeAdmin.t.sol`. This testing easier to parse and see what's missing.